horovod/spark/keras/estimator.py

```diff
@@ -228,8 +229,13 @@ def _get_keras_utils(self):
         if optimizer:
             if isinstance(optimizer, str):
                 pass
-            elif not isinstance(optimizer, tf.keras.optimizers.Optimizer):
-                raise ValueError("optimizer has to be an instance of tensorflow.keras.optimizers.Optimizer")
+            else:
+                if version.parse(tf.keras.__version__) < version.parse("2.11"):
+                    if not isinstance(optimizer, tf.keras.optimizers.Optimizer):
+                        raise ValueError("optimizer has to be an instance of tensorflow.keras.optimizers.Optimizer before Keras 2.11.")
+                else:
+                    if not isinstance(optimizer, tf.keras.optimizers.legacy.Optimizer):
+                        raise ValueError("optimizer has to be an instance of tensorflow.keras.optimizers.legacy.Optimizer starting from Keras 2.11.")
```

Comment:
Can we make the error a bit more verbose?
```suggestion
                if version.parse(tf.keras.__version__) < version.parse("2.11"):
                    if not isinstance(optimizer, tf.keras.optimizers.Optimizer):
                        raise ValueError(f"optimizer has to be an instance of tensorflow.keras.optimizers.Optimizer before Keras 2.11: {type(optimizer)}")
                else:
                    if not isinstance(optimizer, tf.keras.optimizers.legacy.Optimizer):
                        raise ValueError(f"optimizer has to be an instance of tensorflow.keras.optimizers.legacy.Optimizer starting from Keras 2.11: {type(optimizer)}")
```

Reply:
fixed

Reply:
Maybe `{type(optimizer).__name__}` would be better so that

    ValueError: optimizer has to be an instance of tensorflow.keras.optimizers.legacy.Optimizer starting from Keras 2.11: <class 'keras.optimizers.optimizer_experimental.adadelta.Adadelta'>

looks like

    ValueError: optimizer has to be an instance of tensorflow.keras.optimizers.legacy.Optimizer starting from Keras 2.11: keras.optimizers.optimizer_experimental.adadelta.Adadelta